src/Amplitude.Net/AsyncAmplitudeSender.cs

```diff
@@ -29,133 +31,272 @@ public class AsyncAmplitudeSender : IAmplitudeSender,
+        }
+        catch(Exception ex)
+        {
+            _logger.LogError(ex, "Error delivering amplitude identify request");
         }
         finally
         {
-            _concurrencySemaphore.Release();
+            _identifySemaphore.Release();
         }
     }
-
-    private async Task SendRequest(HttpRequestMessage requestMessage)
+    
+    private async Task ProcessEventQueue()
     {
-        using var httpClient = _httpClientFactory.CreateClient();
-        var response = await httpClient.SendAsync(requestMessage);
-        response.EnsureSuccessStatusCode();
+        if (_eventQueue.IsEmpty) return;
+        if (!await _eventSemaphore.WaitAsync(TimeSpan.FromMilliseconds(50))) return;
+            
+        try
+        {
+            const int max = 20;
+            var counter = 0;
+            var payloads = new List<object>();
+
+            while (counter++ < max)
+            {
+                if (!_eventQueue.TryDequeue(out var payload)) break;
+                payloads.Add(payload);
+            }
+
+            if (payloads.Count == 0) return;
+            
+            _logger.LogTrace("Sending Event request");
+            await SendEventRequest(payloads.ToArray());
+            _logger.LogTrace("Event request sent");
+        }
+        catch(Exception ex)
+        {
+            _logger.LogError(ex, "Error delivering amplitude event request");
+        }
+        finally
+        {
+            _eventSemaphore.Release();
+        }
     }
 
-    public ReturnType Identify(IDictionary<string, object?> payload, ILogger logger)
+    private async Task SendIdentifyRequest(object[] payloads)
     {
         const string url = "https://api2.amplitude.com/identify";
 #if NETSTANDARD2_0
-        var identificationBody = JsonConvert.SerializeObject(payload);
+        var identificationBody = JsonConvert.SerializeObject(payloads);
 #else
-        var identificationBody = JsonSerializer.Serialize(payload);
+        var identificationBody = JsonSerializer.Serialize(payloads);
 #endif
         
         var requestBody = new FormUrlEncodedContent(new KeyValuePair<string, string>[]
         {
             new("api_key", _apiKey),
             new("identification", identificationBody)
         });
-        var httpRequest = new HttpRequestMessage(HttpMethod.Post, url);
-        httpRequest.Content = requestBody;
+
+        var count = 0;
+        var delay = TimeSpan.FromSeconds(15);
+        const int MaxRetries = 5;
+        using var httpClient = _httpClientFactory.CreateClient();
+
+        while (count++ < MaxRetries)
+        {
+            var httpRequest = new HttpRequestMessage(HttpMethod.Post, url);
+            httpRequest.Content = requestBody;
+
+            var response = await httpClient.SendAsync(httpRequest);
+
+            if (response.IsSuccessStatusCode)
+            {
+                return;
+            }
+            
+            if((int)response.StatusCode == 429)//too many requests
```

Comment:
Polly for backoffs - maybe later?

Reply:
I didn't see a need to add all the weight of polly here just to implement 5 retries. 15s is a value that comes from amplitude. They say to wait 15s before trying again.